shopinvader_schema_address/schemas.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from extendable_pydantic import ExtendableModelMeta
+from pydantic import BaseModel
+
+from odoo.addons.pydantic import utils
+
+
+class Address(BaseModel, metaclass=ExtendableModelMeta):
+    id: int
+    name: str | None
+    street: str | None
+    street2: str | None
+    zip: str | None
+    city: str | None
+    phone: str | None
+    email: str | None
+    state_id: int | None
+    country_id: int | None
+
+    class Config:
+        orm_mode = True
+        getter_dict = utils.GenericOdooGetter
+
+    @classmethod
+    def from_res_partner(cls, odoo_rec):
+        res = cls.construct()
```

Comment:
construct is going to be renamed in pydantic v2.

Maybe this is better: `cls(id=rec.id, name=rec.name, ...)`